semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -730,6 +730,11 @@ and m_expr a b =
    *)
   | G.Ellipsis _a1, _ -> return ()
   | G.DeepEllipsis (_, a1, _), _b -> m_expr_deep a1 b
+  (* equivalence: extra parentheses equivalence!
+   * must be before literal match below, to allow extra paren in the pattern to still match code without parens
```

Comment:
I'm confused about this comment: aren't you allowing `e` to match `(e)` (aka pattern matches _code_ with extra parens)?